src/client/eppo-client.ts

```diff
@@ -680,7 +680,7 @@ export default class EppoClient {
         expectedVariationType,
       );
 
-      if (!result.variation) {
+      if (!result.variation || result.flagEvaluationDetails.flagEvaluationCode !== 'MATCH') {
```

Comment:
@rasendubi the "catch" is basically a catch all for any error. Fixing that would require a pretty big refactor to how this works. Considering the tests for this have been merged, what do you think about pushing this through and handle that edge case in a separate PR? That way we're not blocked on failing tests if an urgent issue comes up.

Reply:
I'm fine with that although I don't see that as a huge refactor (if `getAssignmentDetail` receives the builder as a parameter instead of creating its own, it's easy to create the builder outside of try-catch here and finish the builder in the catch branch)

Reply:
This didn't turn out to be as big of a refactor as I thought, so I just added them to this PR rather than creating a new one.